src/uct/sm/mm/mm_ep.c

```diff
@@ -113,7 +168,15 @@ static UCS_CLASS_CLEANUP_FUNC(uct_mm_ep_t)
     uct_mm_remote_seg_t *remote_seg;
     struct sglib_hashed_uct_mm_remote_seg_t_iterator iter;
 
-    uct_mm_ep_signal_remote(self, UCT_MM_IFACE_SIGNAL_DISCONNECT);
+    /* don't send a disconnect message (UCT_MM_IFACE_SIGNAL_DISCONNECT) for now
+     * since it may prevent the receiver from progressing and reading incoming messages  */
+
+    /* make sure the slow path function isn't invoked after the ep's cleanup */
+    if (self->cbq_elem_on) {
```

Comment:
pls extract line 175..179 these to function, since it's also used from "uct_mm_ep_signal_remote" after getting success